### PR TITLE
Fix fork repository cycle to self

### LINKED ISSUE #2858: Fork self repository to self
Currently we can fork a repository to the owner himself. Steps:
1. fork user repository to an organization
2. fork organization forked repository to user and change a different name.

image

image

### PR DESCRIPTION
**Pull request by lunny**
Will fix #2858

### CODE REVIEW

File: routers/repo/pull.go [hunk 1]
```diff
@@ -81,6 +80,31 @@ func getForkRepository(ctx *context.Context) *models.Repository {
 			orgs = append(orgs, org)
 		}
 	}
+
+	var testForkRepo = forkRepo
```
Comment (ethantkoenig) [inline routers/repo/pull.go]: nit: please use a more descriptive name than `testForkRepo`. Perhaps `traverseParentRepo`?

Comment (lunny) [reply routers/repo/pull.go]: done.

**Comment by ethantkoenig** [comment]
LGTM

**Comment by lafriks** [comment]
LGTM

**Pull request merged**